practitioner-service/src/main/java/uk/ac/ox/ndph/mts/practitioner_service/controller/PractitionerController.java

```diff
@@ -47,4 +52,14 @@ public PractitionerController(EntityService entityService) {
         String roleAssignmentId = entityService.saveRoleAssignment(roleAssignment);
         return ResponseEntity.status(HttpStatus.CREATED).body(new Response(roleAssignmentId));
     }
+
+    @GetMapping(path = "/roles")
+    @PathParam("identifier")
```

Comment:
it was there before; I'll add it again. like this?

```
    @GetMapping(path = "/{identifier}/roles")
    public ResponseEntity<List<RoleAssignment>> getRoleAssignments(@PathVariable String identifier) {
        if (identifier == null || identifier.isEmpty()) {
            throw new RestException("A required parameter: identifier is blank or missing.");
        }
        List<RoleAssignment> roleAssignments = entityService.getRoleAssignmentsByIdentifier(identifier);
        return  ResponseEntity.ok(roleAssignments);
    }

```

Reply:
are you sure @tamirkamara ? Because according to the rest naming guidelines, it would be correct  if we searched by practitionerId, but identifier is a property within the practitioner and not it's id...

Reply:
Ah, you're right, I was confused with id & identifier. So maybe we should think of another name like "identity" as the business-external name for this property.